include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -67,10 +66,16 @@ namespace seqan3
  * Resolving random access requires logarithmic access into the set and inserting or removing a gap symbol additionally
  * entails updating all subsequent elements in the set to preserve correct cumulative sizes.
  *
+ * ### The seqan3::gap_decorator_anchor_set::iterator type
+ *
+ * \attention The iterator of the seqan3::gap_decorator_anchor_set does not model the std::LegacyInputIterator because
+ * it has no operator->. Note that it does model the std::ranges::InputIterator.
```

Comment:
well, what exactly is a proxy? I thought proxys behave "as if I am working on the actual sequence".. which means I will receive a reference to something I can alter, like for normal iterators. In the gap decorator case, we return a copy of an `gapped<alphabet` character..

Reply:
Well, the proxy behaves like a reference but you return it as a prvalue. The same when you return a copy of gapped<alphabet>. Both of them make the LegacyForwardIterator invalid as it requires a reference to the value type. The operator-> is anyway not used in std::algorithms because they wouldn't know the member variables/functions of the pointed-to object. But they might very well capture the dereferenced result as `auto & ref_elem = *it` which won't work when the iterator returns a prvalue.